### PR TITLE
stellarium: 1.0 -> 1.1

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
followup of https://github.com/NixOS/nixpkgs/pull/194031

**Comment by kilianar** [comment]
As this release fixed the build with Qt 6.4 we might build `stellarium` with Qt 6 instead of Qt 5.

**Comment by kilianar** [comment]
Thanks for updating the pr to build with Qt 6. Should we add `qtwayland` as `buildInput` for wayland support?

On startup `stellarium` doesn't seem to find any time zones, as i get ~25.000 lines like
```shell
StelLocationMgr::loadCitiesBin(): TimeZone for  "Berlin"  not found:  "Europe/Berlin"
StelLocationMgr::loadCitiesBin(): TimeZone for  "Bermejo"  not found:  "America/La_Paz"
StelLocationMgr::loadCitiesBin(): TimeZone for  "Bermeo"  not found:  "Europe/Madrid"
StelLocationMgr::loadCitiesBin(): TimeZone for  "Bern"  not found:  "Europe/Zurich"
StelLocationMgr::loadCitiesBin(): TimeZone for  "Bernau bei Berlin"  not found:  "Europe/Berlin"
```
and dozens of
```shell
Invalid timezone:  "Europe/Berlin"
Invalid timezone:  "Europe/Berlin"
Invalid timezone:  "Europe/Berlin"
Invalid timezone:  "Europe/Berlin"
```
I'm not sure if this is related to packaging or an upstream bug.

File: pkgs/applications/science/astronomy/stellarium/default.nix [hunk 1]
```diff
@@ -19,28 +21,36 @@
...
+    sha256 = "sha256-ellfBZWOkvlRauuwug96C7P/WjQ6dXiDnT0b3KH5zRM=";
   };
 
+  # hotfix for coredump on launch: https://github.com/Stellarium/stellarium/issues/2799
+  patches = [(fetchpatch {
+    url = "https://github.com/Stellarium/stellarium/commit/8cc15658a66230ebfc2346750f443815ae210804.patch";
+    includes = [ "src/core/Dithering.hpp" ];
+    hash = "sha256-OLWfApPV33EvLpduoTFkROC44rHFKM7PSLaVFBZqSqA=";
+  })];
+
```
Comment (kilianar) [inline pkgs/applications/science/astronomy/stellarium/default.nix:39]: Do we need this patch, when building with Qt 6? It does seem to work without the patch. Upstream also only updated the `win-qt5` assets: https://github.com/Stellarium/stellarium/releases/tag/v1.1

**Comment by NickCao** [comment]
> I'm not sure if this is related to packaging or an upstream bug.

Resolved by patching qtbase to load tzdata from the correct location (we did the same for qt5)

**Review by kilianar — APPROVED** [review]

**Pull request merged**